services/Service.go

```diff
@@ -0,0 +1,87 @@
+package services
+
+import (
+	"context"
+	"fmt"
+	"github.com/beevik/etree"
+	"github.com/onego-project/xmlrpc"
+)
+
+// Service structure to make XML-RPC call
+type Service struct {
+	RPC *RPC
+}
+
+// RPC structure represents XML-RPC client and token string
+type RPC struct {
+	Client *xmlrpc.Client
+	Token  string
+}
+
+// enum of result array index
+const (
+	successIndex = iota
+	resultIndex
+	errorCodeIndex
+	idObjectCausedErrorIndex
+)
+
+// UpdateType for replace or merge resource template contents
+type UpdateType int
+
+const (
+	// Replace the whole template
+	Replace UpdateType = iota
+	// Merge new template with the existing one
+	Merge
+)
+
+func (s *Service) call(ctx context.Context, methodName string, args ...interface{}) ([]*xmlrpc.Result, error) {
+	allArgs := append([]interface{}{s.RPC.Token}, args...)
+
+	result, err := s.RPC.Client.Call(ctx, methodName, allArgs...)
+	if err != nil {
+		return nil, err
+	}
+
+	resArr := result.ResultArray()
+	if !resArr[successIndex].ResultBoolean() {
+		if len(resArr) == 4 {
+			return nil, fmt.Errorf("%s, error code: %d, id of the object that caused the error %d",
+				resArr[resultIndex].ResultString(), resArr[errorCodeIndex].ResultInt(), resArr[idObjectCausedErrorIndex].ResultInt())
+		}
+		return nil, fmt.Errorf("%s, code: %d", resArr[resultIndex].ResultString(), resArr[errorCodeIndex].ResultInt())
+	}
+
+	return resArr, nil
+}
+
+// retrieveInfo retrieves information for the given object
+func (s *Service) retrieveInfo(ctx context.Context, methodName string, userID int) (*etree.Document, error) {
```

Comment:
The `userID` parameter should be probably called differently since this method will be used by (maybe) all resource classes.